### PR TITLE
[Do not merge] Adding colmap's SIFT

### PR DESCRIPTION
**Pull request by ayushbaid**
This PR adds Colmap's SIFT by using the python wrapper.

### CODE REVIEW

File: gtsfm/frontend/detector_descriptor/sift_colmap.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import numpy as np
+import pycolmap
+
+from gtsfm.common.image import Image
+from gtsfm.common.keypoints import Keypoints
+from gtsfm.frontend.detector_descriptor.detector_descriptor_base import DetectorDescriptorBase
+
+
+class SIFTColmap(DetectorDescriptorBase):
```
Comment (johnwlambert) [inline gtsfm/frontend/detector_descriptor/sift_colmap.py:17]: looks good Ayush. Can you maybe run the CI on it too to see the benchmarks?

**Comment by ayushbaid** [comment]
I do not have any verified matches from the front-end, even on the door dataset. I'll debug it more tomorrow.

File: environment_linux.yml [hunk 2]
```diff
@@ -39,6 +39,7 @@ dependencies:
   - tabulate
   - pip:
     - pydegensac
+    - pycolmap
```
[the remark this answers is not in the document]
Comment (ayushbaid) [reply environment_linux.yml:42]: The wheels are not supported on Mac right now. So we cannot do that for the time being.

File: gtsfm/frontend/detector_descriptor/sift_colmap.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        Refer to detect() in DetectorBase and describe() in DescriptorBase for details about the output format.
+
+        Args:
+            image: the input image.
+
+        Returns:
+            Detected keypoints, with length N <= max_keypoints.
+            Corr. descriptors, of shape (N, D) where D is the dimension of each descriptor.
+        """
+        vlfeat_keypoints, responses, descriptors = pycolmap.extract_sift(image.value_array)
```
[the remark this answers is not in the document]
Comment (ayushbaid) [reply gtsfm/frontend/detector_descriptor/sift_colmap.py:32]: Thanks. I'll try this.

**Review by adisingh50 — APPROVED** [review]

File: gtsfm/frontend/detector_descriptor/sift_colmap.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        Refer to detect() in DetectorBase and describe() in DescriptorBase for details about the output format.
+
+        Args:
+            image: the input image.
+
+        Returns:
+            Detected keypoints, with length N <= max_keypoints.
+            Corr. descriptors, of shape (N, D) where D is the dimension of each descriptor.
+        """
+        gray_image = image_utils.rgb_to_gray_cv(image)
```
Comment (johnwlambert) [inline gtsfm/frontend/detector_descriptor/sift_colmap.py:33]: could we add a reference to the line number in COLMAP where they do this RGB->gray conversion?

**Comment by ayushbaid** [comment]
No substantial improvement observed. Canning this PR for now.

**Pull request closed without merging**